### ISSUE TITLE
Not piping during add frame

### ISSUE THREAD

**Issue by enricodeleo** [eugeneware/gifencoder#33]
Hi!

I'm creating a gif with 30 frames (but it happens for fewer frames too) and I noticed that even if I'm piping the image encoder to an express response, the image becomes visible only when the encorder.finish is called. Do I have misunderstood the advantage of stream in this use case or is it unexpected?

**Comment by enricodeleo** [comment]
I tried some logging and even if each frame is added. The stream is ok and the browser starts downloading immediately but again, nothing until `finish` is called.

This happens with every kind of gif I tried to create (from flat color to background images) and quality.
The only variable that really matters is how much frames I ask the encoder to add. My understanding is that with stream the browser shouldn't wait for the (for example) 10th frame to start seeing the image.  Am I wrong?

**Comment by heikkipora** [comment]
I was under the impression (too) that browsers would start GIF animation playback immediately, but now I cannot confirm that. It seems that all GIF files i can come up with only start playing after all data has been received.

**Comment by enricodeleo** [comment]
@heikkipora yes it's exactly what I meant before. I cannot understand if this is something I can ever get. It starts being sloppy when you add 30 or 40 frames. As far as you know, is there any way of (at least) make frame addition parallel? Something like gif.js does in the browser with workers?

**Comment by heikkipora** [comment]
Node.js is single-threaded so the answer is no (to being able to process the frames in parallel).

**Comment by enricodeleo** [comment]
@heikkipora, believe me, I know very well about that, that's because I was talking about workers :)
